java-checks/src/main/java/org/sonar/java/checks/unused/UnusedLocalVariableCheck.java

```diff
@@ -112,4 +134,74 @@ private static boolean isTryResource(VariableTree variable) {
     return variable.parent().is(Tree.Kind.LIST) && variable.parent().parent().is(Tree.Kind.TRY_STATEMENT);
   }
 
+  private static Optional<AnalyzerMessage.TextSpan> getQuickFixTextSpan(VariableTree variable) {
+    if (!variable.symbol().usages().isEmpty()) {
+      return Optional.empty();
+    }
+    Tree parent = variable.parent();
+    SyntaxToken lastToken = variable.lastToken();
+    if (parent.is(Tree.Kind.BLOCK, Tree.Kind.INITIALIZER, Tree.Kind.STATIC_INITIALIZER)) {
+      // If the variable is in the declaration list but is not the last one, we also need to include the following comma
+      if (lastToken.text().equals(",")) {
+        return Optional.of(AnalyzerMessage.textSpanBetween(variable.simpleName(), lastToken));
+      }
+      // If the variable is last in the declaration, we need to retrieve the preceding comma
+      Optional<SyntaxToken> precedingComma = getPrecedingComma(variable);
+      if (precedingComma.isPresent()) {
+        AnalyzerMessage.TextSpan value = AnalyzerMessage.textSpanBetween(precedingComma.get(), true, lastToken, false);
+        return Optional.of(value);
+      }
+      return Optional.of(AnalyzerMessage.textSpanBetween(variable.firstToken(), lastToken));
+    } else if (parent.is(Tree.Kind.LIST)) {
+      ListTree<VariableTree> variables = (ListTree<VariableTree>) parent;
+      // If the variable is the only one in the list we can include the entire list
+      if (variables.size() == 1) {
+        return Optional.of(AnalyzerMessage.textSpanFor(variable));
+      }
+      // If the variable is the only one in the list we can include the entire list
+      if (lastToken.text().equals(",")) {
+        return Optional.of(AnalyzerMessage.textSpanBetween(variable.simpleName(), lastToken));
+      }
+      // If the variable is last in the list, we need to retrieve the preceding comma
+      SyntaxToken precedingComma = variables.get(variables.indexOf(variable) - 1).lastToken();
+      return Optional.of(AnalyzerMessage.textSpanBetween(precedingComma, lastToken));
+    } else if (parent.is(Tree.Kind.PATTERN_INSTANCE_OF)) {
+      return Optional.of(AnalyzerMessage.textSpanFor(lastToken));
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<SyntaxToken> getPrecedingComma(VariableTree variable) {
+    Optional<VariableTree> precedingVariable = getPrecedingVariable(variable);
+    if (!precedingVariable.isPresent()) {
+      return Optional.empty();
+    }
+    return Optional.of(precedingVariable.get().lastToken());
+  }
+
+  private static Optional<VariableTree> getPrecedingVariable(VariableTree current) {
+    Tree parent = current.parent();
+    List<Tree> children;
+    if (parent.is(Tree.Kind.BLOCK, Tree.Kind.INITIALIZER)) {
+      children = ((BlockTreeImpl) parent).children();
+    } else {
+      children = ((StaticInitializerTreeImpl) parent).children();
+    }
```

Comment:
Can be simplified because StaticInitializerTreeImpl extends BlockTreeImpl
```suggestion
    List<Tree> children = ((BlockTreeImpl) parent).children();
```